course_content/notebooks/dask_intro.ipynb

```diff
@@ -0,0 +1,258 @@
+    "        dask.multiprocessing.get\n",
+    "- Distributed: Utilises a cluster of distributed machines.\n",
+    "        distributed.Client.get\n",
+    "\n",
+    "#### Worker\n",
+    "A worker receives tasks to process from the scheduler and returns output to the scheduler when that processing is finished. A worker can be a thread, a process or a whole machine, depending on the scheduler used.\n",
+    "\n",
+    "#### Client\n",
+    "A client provides the primary point of access to a distributed scheduler and its associated workers. When using a threaded or multiprocessing scheduler interation with the scheduler is handled by the collection or delayed object.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Task Graph\n",
+    "A graph of tasks (nodes) and the data which is required to pass between them (edges).\n",
+    "In the context of dask:\n",
+    "- Task = circle\n",
+    "- Data = box\n",
+    "- Direction of flow = arrow"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def add(a, b):\n",
+    "    return a + b\n",
+    "\n",
+    "x = 1\n",
+    "y = 2\n",
+    "z = add(x, y)\n",
+    "w = sum([x, y, z])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![](../images/task_graph_def_img_1.png)\n",
+    "\n",
+    "Viewing the relationship between tasks and data can reveal better\\* ways to order the tasks to achieve the same computation.\n",
+    "\n",
+    "\\* Better can mean in less time or using less memory or both!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Dask Graph\n",
+    "Dask stores task graphs in a Python dictionary which maps keys to computations:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "dsk = {'x': 1,\n",
+    "       'y': 2,\n",
+    "       'z': (add, 'x', 'y'),\n",
+    "       'w': (sum, ['x', 'y', 'z'])\n",
+    "      }"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Dask is different from other parallelising libraries in that it uses ordinary Python structures to represent task graphs instead of a specialised API:\n",
+    "\n",
+    "- {Dicts}\n",
+    "- (Tuples)\n",
+    "- Functions()\n",
+    "- Values\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Exercise: Terminology\n",
+    "\n",
+    "Now it is time to test what you have learned! Answer the following by matching the definitions with the terms."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "Q: What is a scheduler? (Answer in the cell below.)"
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
You could always make this cell_type raw nbconvert so that when people start to type it doesn't expect code
But I may be being a bit fussy